system/lib/dlmalloc.c

```diff
@@ -3020,7 +3020,7 @@ I = (K << 1) + ((S >> (K + (TREEBIN_SHIFT-1)) & 1));\
 
 /* index corresponding to given bit. Use x86 asm if possible */
 
-#if defined(__GNUC__) && (defined(__i386__) || defined(__x86_64__))
+#if defined(__GNUC__) && (defined(__i386__) || defined(__x86_64__) || defined(__EMSCRIPTEN__))
```

Comment:
I wonder this doesn't use `#if __has_builtin(__builtin_ctz)`  .. I guess maybe dlmalloc was written before such a macro existed?

I also wonder if we should use `__wasm__` here instead of `__EMSCRIPTEN__`?  I guess it doesn't matter much since if there is almost no chance of upstreaming these chagnes.

Reply:
Yeah, `__wasm__` probably is a better choice; I used `__EMSCRIPTEN__` since that was used by other modifications in that file.